Adds 5 new scroll items with unique effects

someaddons opened the pull request:
Closes #6091

# Changes proposed in this pull request:
Adds 5 new scroll items with unique effects:
A colony home teleportation scroll, which teleports the user to the registered colony. 
A colony home area teleportation scroll which teleports the user and nearby players to the colony.
A regeneration scroll, granting regeneration to the player and citizens around him, found in chests.
A guard help scroll which temporarily summons the guards of a registered guard building to the user on follow mode.
A worker highlight scroll which can be used on a worker building an highlights the worker with a glow effect and gives him a slight speed effect too.
![image](https://user-images.githubusercontent.com/38401808/100540763-ae16f280-323f-11eb-9547-7583bb17bbf0.png)


Fix some of our sounds beeing stereo and not playing positioned
Review please

someaddons committed:
Adds 5 new scroll items with unique effects:
A colony home teleportation scroll, which teleports the user to the registered colony. Made by the enchanter
A colony home area teleportation scroll which teleports the user and nearby players to the colony. Made by the enchanter
A regeneration scroll, granting regeneration to the player and citizens around him, found in chests.
A guard help scroll which temporarily summons the guards of a registered guard building to the user on follow mode.
A worker highlight scroll which can be used on a worker building an highlights the worker with a glow effect and gives him a slight speed effect too.

Fix some of our sounds beeing stereo and not playing positioned

someaddons committed:
javadoc

ravenbuilder934 commented:
> Fix some of our sounds beeing stereo and not playing positioned

Does this close #6091?

ravenbuilder934 commented on src/main/resources/assets/minecolonies/lang/en_us.json:
```diff
@@ -338,6 +338,33 @@
+  "item.minecolonies.scroll.colony.none": "None, set one with sneak rightclick",
+  "item.minecolonies.scroll_area_tp": "Ultrasafe colony group-teleport scroll",
+  "item.minecolonies.scroll_area_tp.tip": "Teleport yourself and nearby players to the colony",
+  "item.minecolonies.scroll_buff": "Sacred scroll of Regeneration",
```
```suggestion
  "item.minecolonies.scroll_buff": "Sacred Scroll of Regeneration",
```

ravenbuilder934 commented on src/main/resources/assets/minecolonies/lang/en_us.json:
```diff
@@ -338,6 +338,33 @@
+  "item.minecolonies.scroll_tp": "Ultrasafe colony teleport scroll",
+  "item.minecolonies.scroll_tp.tip": "Safely teleports you to the set colony",
+  "item.minecolonies.scroll.colony.tip": "Colony: %s",
+  "item.minecolonies.scroll.colony.none": "None, set one with sneak rightclick",
```
```suggestion
  "item.minecolonies.scroll.colony.none": "None, set one with sneak + right-click",
```

ravenbuilder934 commented on src/main/resources/assets/minecolonies/lang/en_us.json:
```diff
@@ -338,6 +338,33 @@
+  "item.minecolonies.scroll_area_tp": "Ultrasafe colony group-teleport scroll",
+  "item.minecolonies.scroll_area_tp.tip": "Teleport yourself and nearby players to the colony",
+  "item.minecolonies.scroll_buff": "Sacred scroll of Regeneration",
+  "item.minecolonies.scroll_buff.tip": "Applies regeneration to citizens around you, at least most of the times...",
```
```suggestion
  "item.minecolonies.scroll_buff.tip": "Applies regeneration to citizens around you, at least most of the time...",
```

ravenbuilder934 commented on src/main/resources/assets/minecolonies/lang/en_us.json:
```diff
@@ -338,6 +338,33 @@
+  "item.minecolonies.scroll_highlight": "Worker-where-are-you scroll",
+  "item.minecolonies.scroll_highlight.tip": "Sneak-Use on a minecolonies building, to highlight and speed up its workers",
+  "minecolonies.scroll.nopermission": "You do not have the required permissions for this colony.",
+  "minecolonies.scroll.needcolony": "Please register this to a colony first with shift-rightclicking a building.",
```
```suggestion
  "minecolonies.scroll.needcolony": "Please register this to a colony first with shift + right-clicking a building.",
```

ravenbuilder934 commented on src/main/resources/assets/minecolonies/lang/en_us.json:
```diff
@@ -338,6 +338,33 @@
+  "item.minecolonies.scroll_highlight.tip": "Sneak-Use on a minecolonies building, to highlight and speed up its workers",
+  "minecolonies.scroll.nopermission": "You do not have the required permissions for this colony.",
+  "minecolonies.scroll.needcolony": "Please register this to a colony first with shift-rightclicking a building.",
+  "minecolonies.scroll.noguardbuilding": "Please register this to a guard building by shift-rightclicking it.",
```
```suggestion
  "minecolonies.scroll.noguardbuilding": "Please register this to a guard building by shift + right-clicking it.",
```

ravenbuilder934 commented on src/main/resources/assets/minecolonies/lang/en_us.json:
```diff
@@ -338,6 +338,33 @@
+  "minecolonies.scroll.wrongdim": "Can only be used in the same dimension as the colony.",
+  "minecolonies.scroll.registered": "Registered scroll with colony %s",
+  "minecolonies.scroll.failed1": "The unstable magic of the scroll backfired!",
+  "minecolonies.scroll.failed2": "Whoops, what happened now?!",
```
```suggestion
  "minecolonies.scroll.failed2": "Oopsies!",
```

ravenbuilder934 commented on src/main/resources/assets/minecolonies/lang/en_us.json:
```diff
@@ -338,6 +338,33 @@
+  "minecolonies.scroll.registered": "Registered scroll with colony %s",
+  "minecolonies.scroll.failed1": "The unstable magic of the scroll backfired!",
+  "minecolonies.scroll.failed2": "Whoops, what happened now?!",
+  "minecolonies.scroll.failed3": "This scroll needs to go back to the drawing board!",
```
```suggestion
  "minecolonies.scroll.failed3": "Uh oh!",
```

ravenbuilder934 commented on src/main/resources/assets/minecolonies/lang/en_us.json:
```diff
@@ -338,6 +338,33 @@
+  "minecolonies.scroll.failed1": "The unstable magic of the scroll backfired!",
+  "minecolonies.scroll.failed2": "Whoops, what happened now?!",
+  "minecolonies.scroll.failed3": "This scroll needs to go back to the drawing board!",
+  "minecolonies.scroll.failed4": "The scroll doesnt seem to be made by the best magician out there!",
```
```suggestion
  "minecolonies.scroll.failed4": "Sorry about that!",
```

ravenbuilder934 commented on src/main/resources/assets/minecolonies/lang/en_us.json:
```diff
@@ -338,6 +338,33 @@
+  "minecolonies.scroll.failed4": "The scroll doesnt seem to be made by the best magician out there!",
+  "minecolonies.scroll.failed5": "Well, this is embarrassing.",
+  "minecolonies.scroll.failed6": "Looks like there's a typo in one of the ancient glyphs!",
+  "minecolonies.scroll.failed7": "Did my I get the wrong spell again?!",
```
```suggestion
  "minecolonies.scroll.failed7": "Did my creator copy the wrong spell again?!",
```

ravenbuilder934 commented on src/main/resources/assets/minecolonies/lang/en_us.json:
```diff
@@ -338,6 +338,33 @@
+  "minecolonies.scroll.failed5": "Well, this is embarrassing.",
+  "minecolonies.scroll.failed6": "Looks like there's a typo in one of the ancient glyphs!",
+  "minecolonies.scroll.failed7": "Did my I get the wrong spell again?!",
+  "minecolonies.scroll.failed8": "Wasn't this supposed to do something?",
```
```suggestion
  "minecolonies.scroll.failed8": "Oh, was I supposed to do something?",
```

ravenbuilder934 commented on src/main/resources/assets/minecolonies/lang/en_us.json:
```diff
@@ -338,6 +338,33 @@
+  "minecolonies.scroll.failed7": "Did my I get the wrong spell again?!",
+  "minecolonies.scroll.failed8": "Wasn't this supposed to do something?",
+  "minecolonies.scroll.failed9": "404 Spell Not Found",
+  "minecolonies.scroll.failed10": "Maybe try some ladders instead?",
```
```suggestion
  "minecolonies.scroll.failed10": "Maybe try some ladders instead?",
  "minecolonies.scroll.failed11": "I blame Someaddons.",
  "minecolonies.scroll.failed12": "I blame Wissi.",
  "minecolonies.scroll.failed13": "418 I'm a teapot, not a spell!",
  "minecolonies.scroll.failed14": "You should try our sister game, ColonyColonyColonyColonyColonyColonyColonyColonyColony!",
```

ravenbuilder934 commented on src/main/java/com/minecolonies/coremod/items/AbstractItemScroll.java:
```diff
@@ -0,0 +1,167 @@
+{
+    public static final String TAG_COLONY_DIM       = "colony_dim";
+    public static final String TAG_BUILDING_POS     = "building_pos";
+    public static final int    FAIL_RESPONSES_TOTAL = 10;
```
```suggestion
    public static final int    FAIL_RESPONSES_TOTAL = 14;
```

someaddons committed:
Update src/main/resources/assets/minecolonies/lang/en_us.json

Co-authored-by: ravenbuilder934 <sabrinajamie@outlook.com>

someaddons committed:
Update src/main/resources/assets/minecolonies/lang/en_us.json

Co-authored-by: ravenbuilder934 <sabrinajamie@outlook.com>

someaddons committed:
Update src/main/resources/assets/minecolonies/lang/en_us.json

Co-authored-by: ravenbuilder934 <sabrinajamie@outlook.com>

someaddons committed:
Update src/main/resources/assets/minecolonies/lang/en_us.json

Co-authored-by: ravenbuilder934 <sabrinajamie@outlook.com>

someaddons committed:
Update src/main/resources/assets/minecolonies/lang/en_us.json

Co-authored-by: ravenbuilder934 <sabrinajamie@outlook.com>

someaddons committed:
Update src/main/resources/assets/minecolonies/lang/en_us.json

Co-authored-by: ravenbuilder934 <sabrinajamie@outlook.com>

marchermans commented on src/main/java/com/minecolonies/coremod/colony/buildings/workerbuildings/BuildingEnchanter.java:
```diff
@@ -33,7 +34,7 @@
  * The enchanter building.
  */
-public class BuildingEnchanter extends AbstractBuildingWorker
+public class BuildingEnchanter extends AbstractBuildingCrafter
```
This building was turned into a crafter. Yet no methods where overriden. Sooo is this a default crafter? I think you would like ant to limit the recipes would you not?

Raycoms replied:
@OrionDevelopment afaik that is all data driven now. @Mekle001 this should be fine as is, right?

someaddons replied:
hm good point I'll go ahead and test it

someaddons replied:
was able to teach it sticks xD

someaddons committed:
Small statemachine improvement (#6171)

This changes the statemachine to no longer do hash lookups frequently when it already does know/has a reference to the result. Instead we do the lookup once the state changes and keep the reference to the current state's transitions.
Removed streams in the basic statemachine aswell.

Raycoms committed:
fix #6161 (#6167)

Performance improvement
Fix free blocks config
Fix abandoned colony calc
Fix infinite request loops
Fix client side desyncs of racks

someaddons committed:
disallow adding recipes
